fluent-plugin-enhance-k8s-metadata/test/sumologic/kubernetes/test_reader.rb

```diff
@@ -49,5 +49,14 @@ def log
     metadata = fetch_pod_metadata('non-exist', 'somepod')
     assert_not_nil metadata
     assert_equal 0, metadata.size
+    assert log.logs.any? { |log| log.include?('404') }
+  end
+
+  test 'fetch_pod_metadata for pod with non-existent owner logs warning' do
+    metadata = fetch_pod_metadata('sumologic', 'pod-with-nonexistent-owner')
+    assert_not_nil metadata
+    assert_equal '1691804714', metadata['pod_labels']['pod_labels']['pod-template-hash']
+    assert_empty metadata['owners']
+    assert log.logs.any? { |log| log.include?('failed to fetch resource') }, "'failed to fetch resource' not found in logs: #{log.logs}"
```

Comment:
can we test for full message here?

Reply:
Yes we can, I considered this but wanted to make it as concise as possible

Reply:
I think we should test it. It can be another test

Reply:
Please see change in https://github.com/SumoLogic/sumologic-kubernetes-collection/pull/1212/commits/84b9cfb349302d1f7aeff336034d8074486d30cb - is this what you had in mind @sumo-drosiek?

Reply:
Yes, thanks